exporter/elasticsearchexporter/exporter_test.go

```diff
@@ -1932,6 +1935,147 @@ func TestExporterTraces(t *testing.T) {
 	})
 }
 
+func TestExporter_MappingModeMetadata(t *testing.T) {
+	otelContext := client.NewContext(context.Background(), client.Info{
+		Metadata: client.NewMetadata(map[string][]string{"X-Elastic-Mapping-Mode": {"otel"}}),
+	})
+	ecsContext := client.NewContext(context.Background(), client.Info{
+		Metadata: client.NewMetadata(map[string][]string{"X-Elastic-Mapping-Mode": {"ecs"}}),
+	})
+	noneContext := client.NewContext(context.Background(), client.Info{
+		Metadata: client.NewMetadata(map[string][]string{"X-Elastic-Mapping-Mode": {"none"}}),
+	})
+
+	logs := plog.NewLogs()
+	resourceLog := logs.ResourceLogs().AppendEmpty()
+	resourceLog.Resource().Attributes().PutStr("k", "v")
+	scopeLog := resourceLog.ScopeLogs().AppendEmpty()
+	scopeLog.LogRecords().AppendEmpty()
+	logs.MarkReadOnly()
+
+	metrics := pmetric.NewMetrics()
+	resourceMetrics := metrics.ResourceMetrics().AppendEmpty()
+	resourceMetrics.Resource().Attributes().PutStr("k", "v")
+	metric := resourceMetrics.ScopeMetrics().AppendEmpty().Metrics().AppendEmpty()
+	metric.SetName("metric.foo")
+	metric.SetEmptySum().DataPoints().AppendEmpty().SetIntValue(123)
+	metrics.MarkReadOnly()
+
+	traces := ptrace.NewTraces()
+	resourceSpans := traces.ResourceSpans().AppendEmpty()
+	resourceSpans.Resource().Attributes().PutStr("k", "v")
+	resourceSpans.ScopeSpans().AppendEmpty().Spans().AppendEmpty()
+	traces.MarkReadOnly()
+
+	setAllowedMappingModes := func(cfg *Config) {
+		cfg.Mapping.AllowedModes = []string{"ecs", "otel"}
+	}
+
+	checkOTelResource := func(t *testing.T, doc []byte, _ string) {
+		t.Helper()
+		assert.JSONEq(t, `{"k":"v"}`, gjson.GetBytes(doc, `resource.attributes`).Raw)
+	}
+	checkECSResource := func(t *testing.T, doc []byte, signal string) {
+		t.Helper()
+		if signal == "traces" {
+			// ecs mode schema for spans is currently very different
+			// to logs and metrics
+			assert.Equal(t, "v", gjson.GetBytes(doc, "Resource.k").Str)
+		} else {
+			assert.Equal(t, "v", gjson.GetBytes(doc, "k").Str)
+		}
+	}
+
+	testcases := []struct {
+		name      string
+		ctx       context.Context
+		check     func(_ *testing.T, doc []byte, signal string)
+		expectErr string
+	}{{
+		name:  "otel",
+		ctx:   otelContext,
+		check: checkOTelResource,
+	}, {
+		name:  "ecs",
+		ctx:   ecsContext,
+		check: checkECSResource,
+	}, {
+		name:      "bodymap",
+		ctx:       noneContext,
+		expectErr: `unsupported mapping mode "none", expected one of ["ecs" "otel"]`,
+	}}
+
+	t.Run("logs", func(t *testing.T) {
+		for _, tc := range testcases {
+			t.Run(tc.name, func(t *testing.T) {
+				rec := newBulkRecorder()
+				server := newESTestServer(t, func(docs []itemRequest) ([]itemResponse, error) {
+					rec.Record(docs)
+					return itemsAllOK(docs)
```

Comment:
is there a way to quickly check if the supplied mapping mode is respected?

Reply:
That's what the check function is doing -- it's verifying the document structure is as expected for the given mapping mode.